ksqldb-common/src/main/java/io/confluent/ksql/metrics/StreamsErrorCollector.java

```diff
@@ -38,36 +39,29 @@
   public static final String CONSUMER_FAILED_MESSAGES_PER_SEC
       = "consumer-failed-messages-per-sec";
 
-  private static final Map<String, StreamsErrorCollector> COLLECTORS = Maps.newConcurrentMap();
-
-  private final Map<String, TopicSensors<Object>> topicSensors = Maps.newConcurrentMap();
-  private String id;
+  private final MetricCollectors metricCollectors;
   private final Metrics metrics;
+  private final Map<String, TopicSensors<Object>> topicSensors = Maps.newConcurrentMap();
+  private final String id;
   private final Time time;
 
-  private StreamsErrorCollector() {
-    this.metrics = MetricCollectors.getMetrics();
-    this.time = MetricCollectors.getTime();
-  }
-
-  public void setId(final String id) {
-    this.id = id;
-  }
-
-  private void collect(final String topic) {
-    topicSensors
-        .computeIfAbsent(topic, this::buildSensors)
-        .increment(null, true);
+  @SuppressFBWarnings(
+      value = "EI_EXPOSE_REP2",
+      justification = "metrics"
+  )
+  public StreamsErrorCollector(final String applicationId, final MetricCollectors collectors) {
+    this.metricCollectors = collectors;
+    this.metrics = collectors.getMetrics();
+    this.time = collectors.getTime();
+    this.id = collectors.addCollector(applicationId, this);
```

Comment:
That's a much better idea. I'm not sure what I was thinking. I'll fix that in a follow-up (to avoid waiting for ITs again) unless there's another change I have to do in this PR.